File: python/strategy/DuckSimulator/behavior/FlyBehavior.py

class FlyBehavior:

    def fly(self):
        return

class FlyWithWings(FlyBehavior):

    def __init__(self, duck_name):
        self.__duck_name = duck_name

    def fly(self):
        print(self.__duck_name + " is flying wih wings")

class FlyNoWay(FlyBehavior):

    def __init__(self, duck_name):
        self.__duck_name = duck_name

    def fly(self):
        print(self.__duck_name + " is not flying")

File: C++/Strategy/DuckSimulator/behavior/FlyBehavior.cpp
#include "iostream"
#include <string>
using namespace std;

class FlyBehavior {

public:
    virtual void fly() = 0;
};

class FlyWithWings : public FlyBehavior {

public:
    FlyWithWings(string duckName) : duckName(duckName) {};
    void fly() override { cout << duckName << " is flying" << endl; };

private:
    string duckName;
};

class FlyNoWay : public FlyBehavior {

public:
    FlyNoWay(string duckName) : duckName(duckName) {};
    void fly() override { cout << duckName << " is not flying" << endl; };

private:
    string duckName;
};

